### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by TheSuperiorStanislav**

### CODE REVIEW

File: tests/test_auto_schema.py [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+        assert getattr(model, field) == getattr(test_model, field)
+        if "nullable" not in field and "property" not in field:
+            with pytest.raises(pydantic.ValidationError):
+                setattr(model, field, None)
+
+
+async def test_auto_schema_type_override_generation(
+    test_model: models.TestModel,
+) -> None:
+    """Test that in auto schema generation ypu can override type."""
```
Comment (MrShirokow) [inline tests/test_auto_schema.py:68]:
```suggestion
    """Test that in auto schema generation you can override type."""
```

File: tests/test_auto_schema.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    schema = AutoSchema.get_schema()
+    model = schema.model_validate(test_model)
+    for field in AutoSchema.Meta.fields:
+        assert getattr(model, field) == getattr(test_model, field)
+        if "nullable" not in field and "property" not in field:
+            with pytest.raises(pydantic.ValidationError):
+                setattr(model, field, None)
+
+
+async def test_auto_schema_type_override_generation(
```
Comment (MrShirokow) [inline tests/test_auto_schema.py:65]: For me, it's unclear, what this test checks...
Can you extend docs please?
Or maybe need rename for test

File: tests/test_factories.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        ],
+    ):
+        """Factory to generate TestModel."""
+
+        class Meta:
+            model = models.TestModel
+
+    with pytest.raises(
+        ValueError,
+        match="Repository class in not set in Meta class",
```
Comment (MrShirokow) [inline tests/test_factories.py:26]:
```suggestion
        match="Repository class is not set in Meta class",
```

File: tests/test_repositories.py [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+    assert new_repo.db_session is repository.db_session
+
+
+async def test_soft_delete(
+    soft_delete_test_model: models.SoftDeleteTestModel,
+    soft_delete_repository: repositories.SoftDeleteTestModelRepository,
+) -> None:
+    """Test soft deletion."""
+    await soft_delete_repository.delete(soft_delete_test_model)
+    deleted = await soft_delete_repository.get(soft_delete_test_model.id)
```
Comment (MrShirokow) [inline tests/test_repositories.py:26]:
```suggestion
    soft_deleted_object = await soft_delete_repository.get(soft_delete_test_model.id)
```

**Review by MrShirokow — COMMENTED** [review]
Please extend docs for some tests (especially for `auto scheme) because sometimes test's behaviour is not unclear

File: tests/models.py [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+        sqlalchemy.Boolean(),
+        nullable=False,
+    )
+
+    true_false_nullable: sqlalchemy.orm.Mapped[bool | None] = (
+        sqlalchemy.orm.mapped_column(
+            sqlalchemy.Boolean(),
+            nullable=True,
+        )
+    )
```
Comment (LeonidMalin) [inline tests/models.py:142]:
```suggestion
    boolean: sqlalchemy.orm.Mapped[bool] = sqlalchemy.orm.mapped_column(
        sqlalchemy.Boolean(),
        nullable=False,
    )

    boolean_nullable: sqlalchemy.orm.Mapped[bool | None] = (
        sqlalchemy.orm.mapped_column(
            sqlalchemy.Boolean(),
            nullable=True,
        )
    )
```

**Review by LeonidMalin — APPROVED** [review]

File: tests/factories.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        ],
+    )
+    date_time = factory.Faker("date_time")
+    date = factory.Faker("date_between")
+    timedelta = factory.Faker("time_delta")
+
+    class Meta:
+        model = models.TestModel
+        repository = repositories.TestModelRepository
+        sub_factories: typing.ClassVar = {
```
[the remark this answers is not in the document]
Comment (TheSuperiorStanislav) [reply tests/factories.py:50]: Hmm, mypy is not complaning so I think we can leave as it is

File: tests/factories.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+            "related_model": "tests.factories.RelatedModelFactory",
+            "related_models": ("tests.factories.RelatedModelFactory", 5),
+        }
+
+
+class SoftDeleteTestModelFactory(
```
[the remark this answers is not in the document]
Comment (TheSuperiorStanislav) [reply tests/factories.py:56]: Why? It should be None by default

**Review by MrShirokow — APPROVED** [review]

**Pull request merged**